src/zip.c

```diff
@@ -277,7 +277,14 @@ int zip_entry_open(struct zip_t *zip, const char *entryname) {
   zip->entry.header_offset = zip->archive.m_archive_size;
   memset(zip->entry.header, 0, MZ_ZIP_LOCAL_DIR_HEADER_SIZE * sizeof(mz_uint8));
   zip->entry.method = 0;
+
+  // UNIX or APPLE
+#if MZ_PLATFORM == 3 || MZ_PLATFORM == 19
```

Comment:
maybe make it compatible with what we have in read and extract, where on windows _do nothing_ else:
`(mz_uint32)((0100644 & 0xFFFF) << 16);`

Reply:
Yes.